### PR TITLE
#84 Send email notification when account details are changed

### PR DESCRIPTION
**Pull request by Rita-C**

### CODE REVIEW

File: grails-app/services/au/org/ala/userdetails/EmailService.groovy [hunk 1]
```diff
@@ -42,19 +42,31 @@ class EmailService {
...
           body (view: '/email/activateAccount',
                 plugin:"email-confirmation",
-                model:[link:getServerUrl() + "activateAccount/" + user.id + "/"  + authKey, orgNameLong: grailsApplication.config.skin.orgNameLong ]
+                model:[userName: user.firstName, link: getServerUrl() + "activateAccount/" + user.id + "/"  + authKey, orgNameLong: grailsApplication.config.skin.orgNameLong ]
           )
         }
     }
 
-    def sendAccountActivationSuccess(def user, def activatedAlerts) {
+    def sendAccountActivationSuccess(user, activatedAlerts) {
```
Comment (nickdos) [inline grails-app/services/au/org/ala/userdetails/EmailService.groovy:50]: If modifying, then I suggest adding (proper) types to the args (and return type while you're at it). Services should be strongly typed IMHO.

File: grails-app/services/au/org/ala/userdetails/EmailService.groovy [hunk 2]
```diff
@@ -42,19 +42,31 @@ class EmailService {
...
             body (view: '/email/activateAccountSuccess',
                     plugin:"email-confirmation",
-                    model:[activatedAlerts: activatedAlerts, alertsUrl: grailsApplication.config.alerts.url]
+                    model:[userName: user.firstName, activatedAlerts: activatedAlerts, alertsUrl: grailsApplication.config.alerts.url]
+            )
+        }
+    }
+
+    def sendUpdateProfileSuccess(user) {
+        sendMail {
```
Comment (nickdos) [inline grails-app/services/au/org/ala/userdetails/EmailService.groovy:63]: Sending mail is a slightly error-prone activity, so I'd be inclined to add exception handling. In particular servers can be deployed with missing or incorrect mail settings, so worthwhile catching this early - and giving user a meaningful error message (as opposed to generic 500 page that they would likely get with implicit exception handling from Grails controllers). See https://github.com/AtlasOfLivingAustralia/userdetails/blob/dev/grails-app/services/au/org/ala/userdetails/EmailService.groovy#L23

File: grails-app/views/email/resetPassword.gsp [hunk 3]
```diff
@@ -3,6 +3,7 @@
   <head><title>${emailTitle}</title></head>
   <body>
     <h3>${emailTitle}</h3>
+    <p>Dear ${userName}</p>
```
Comment (nickdos) [inline grails-app/views/email/resetPassword.gsp:6]: `Dear` should be inside a `g:message`

**Review by nickdos — COMMENTED** [review]
A few comments added.

File: grails-app/services/au/org/ala/userdetails/EmailService.groovy (hunk 1 above)
Comment (Rita-C) [reply grails-app/services/au/org/ala/userdetails/EmailService.groovy:50]: This is the only method has `def` "type, I was simply making methods signature consistent.

File: grails-app/services/au/org/ala/userdetails/EmailService.groovy (hunk 2 above)
Comment (Rita-C) [reply grails-app/services/au/org/ala/userdetails/EmailService.groovy:63]: can do

File: grails-app/views/email/resetPassword.gsp (hunk 3 above)
Comment (Rita-C) [reply grails-app/views/email/resetPassword.gsp:6]: good point

**Comment by Rita-C** [comment]
> A few comments added.

updated

File: grails-app/services/au/org/ala/userdetails/EmailService.groovy (hunk 1 above)
Comment (nickdos) [reply grails-app/services/au/org/ala/userdetails/EmailService.groovy:54]: NW can leave as is - consistent is good. Including types in services is one of my pet peeves ;-)

**Review by nickdos — APPROVED** [review]

**Pull request merged**